samples/hello_world/src/main.c

```diff
@@ -4,47 +4,134 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <stdio.h>
 #include <zephyr/kernel.h>
 #include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/flash.h>
 #include <scu.h>
 
+#define ATTR_INF	"\x1b[32;1m" // ANSI_COLOR_GREEN
+#define ATTR_ERR	"\x1b[31;1m" // ANSI_COLOR_RED
+#define ATTR_RST	"\x1b[37;1m" // ANSI_COLOR_RESET
+
+#define FLASH_RW_SIZE 255
+
+void Flash_Test(const struct device *flash, uint32_t FLASH_ADDR, uint8_t EVEN_ODD_MUL, uint8_t FORMATTER) {
+	int errorcode = 0;
+	uint8_t flash_data_write[FLASH_RW_SIZE] = {0};
+	uint8_t flash_data_read[FLASH_RW_SIZE] = {0};
+	errorcode = flash_read(flash, FLASH_ADDR, flash_data_read, FLASH_RW_SIZE);
```

Comment:
I dont see any code where the flash_read is assigned some driver function. Where is the code / configuration that ties the flash_read to the espi / spi controller driver code. Also is the flash in this case read in memory mapped mode? Is the memory mapped mode supported? Have we tested it?

Reply:
flash_read is a generic API which is bound to spi_nor.c driver code for nor spi flash devices. Inside the driver read function there are calls which end up in andestech atcspi200 driver.

This is not a memory mapped read actually. I did not see in the driver explicit memory map reading being enabled. The same was the case in our driver of EmbeddedCommon-Dev. Jorge confirmed to me that it was by default memory mapped mode. 

I think the same could be the case here. I will try to do memory mapped reading once to confirm this. Shall we need memory mapped reading in Zephyr?

Reply:
> flash_read is a generic API which is bound to spi_nor.c driver code for nor spi flash devices. Inside the driver read function there are calls which end up in andestech atcspi200 driver.
> 
> This is not a memory mapped read actually. I did not see in the driver explicit memory map reading being enabled. The same was the case in our driver of EmbeddedCommon-Dev. Jorge confirmed to me that it was by default memory mapped mode.
> 
> I think the same could be the case here. I will try to do memory mapped reading once to confirm this. Shall we need memory mapped reading in Zephyr?

It may be a good idea to enable memory mapped read. The customer application code may use memory mapped read. Also if we do end up adding a command line, that could also end up using a memory mapped read. If you can just test it, that should be enough. I was thinking test in command line, but if we dont have enough space for commands, we could test it in the application in memory mode. In fact that is easier, we just need to access that memory location

Reply:
Indeed, reading it in an application is much easier. I will check it tomorrow.